crates/client/src/client.rs

```diff
@@ -938,8 +938,10 @@ impl FuelClient {
     pub async fn submit_and_await_status<'a>(
         &'a self,
         tx: &'a Transaction,
+        allow_preconfirmation: bool,
     ) -> io::Result<impl Stream<Item = io::Result<TransactionStatus>> + 'a> {
-        self.submit_and_await_status_opt(tx, None).await
+        self.submit_and_await_status_opt(tx, None, Some(allow_preconfirmation))
```

Comment:
You also forgot to update `subscribe_transaction_status` function. Maybe instead of updating it, we could add a new `subscribe_transaction_status_opt`

Reply:
Done.